### PR TITLE
Fixes #1270. Construct correct Raycaster

### LINKED ISSUE #1270: map.intersectMapObjects(e.pageX, e.pageY) notWork
**Describe the bug**
after adding litener for click to [examle](https://www.harp.gl/docs/master/examples/#datasource_features_lines-and-points.html)
`
map.addDataSource(featuresDataSource).then(() => {
        map.canvas.addEventListener("click", (e: MouseEvent) => {
            const intersectionResults = map.intersectMapObjects(e.pageX, e.pageY);
            const usableResults = intersectionResults.filter(result => result.userData !== undefined);
            console.log(intersectionResults, usableResults)
        });
    });
`
has [error](http://tinyurl.com/yx89to8v) in console

### PR DESCRIPTION
**Pull request by nzjony**
Fixes #1270

### CODE REVIEW

File: @here/harp-mapview/lib/MapView.ts [hunk 1]
```diff
@@ -749,7 +750,7 @@ export class MapView extends THREE.EventDispatcher {
     private m_enablePolarDataSource: boolean = true;
 
     // gestures
-    private readonly m_raycaster = new THREE.Raycaster();
+    private readonly m_raycaster = new PickingRaycaster(this);
```
Comment (ninok) [inline @here/harp-mapview/lib/MapView.ts]: I would prefer to get rid of this `PickingRaycaster` or at least not keep the whole `MapView` reference b/c as we know from history people will do bad things if the have access to the whole `MapView`.
As far as I can see, the `PickingRaycaster` is only used in `MapViewPoints` and there we use the camera and the canvas size. Would be cleaner to add these properties explicitly and we would also avoid to query the DOM for every pick event.
Long story short: I would prefer if you do the changes now but if you don't have time, please create a ticket.

Comment (nzjony) [reply @here/harp-mapview/lib/MapView.ts]: I will fix now, seems to be a small change.

File: @here/harp-mapview/lib/PickingRaycaster.ts [hunk 2]
```diff
@@ -5,21 +5,21 @@
...
+ * space, used for example in the [[MapViewPoints]].
  */
 export class PickingRaycaster extends THREE.Raycaster {
     /**
-     * Constructs a `MapViewRaycaster`. It keeps a reference to [[MapView]] in order to access the
-     * active camera when calling the custom method `raycastPoints`.
+     * Constructor.
      *
-     * @param m_mapView the active [[MapView]].
+     * @param camera the camera used for projection to screen.
```
Comment (ninok) [inline @here/harp-mapview/lib/PickingRaycaster.ts]: Maybe you can explain a bit more that this camera is the world camera (i.e. equal to `MapView.camera`) whereas the `Raycaster.camera` is our internal RTE camera (i.e. equal to `MapView.m_rteCamera`).
Maybe it also makes sense to rename camera to worldCamera b/c otherwise we have two properties with the same name `camera` (I wonder that typescript is not complaining abut this)

Comment (nzjony) [reply @here/harp-mapview/lib/PickingRaycaster.ts]: I have fixed the PickingRaycaster to take the rteCamera.

File: @here/harp-mapview/lib/PickingRaycaster.ts [hunk 3]
```diff
@@ -5,21 +5,22 @@
...
+     * Constructor.
      *
-     * @param m_mapView the active [[MapView]].
+     * @param camera the camera used for projection to screen. Note, this must use the camera used
+     * to finally render the scene and not the camera in world space.
+     * @param width the canvas width.
+     * @param height the canvas height.
      */
-    constructor(public mapView: MapView) {
+    constructor(public camera: THREE.Camera, public width: number, public height: number) {
```
Comment (ninok) [inline @here/harp-mapview/lib/PickingRaycaster.ts]: Why do we still need the camera as a member? You should just forward the camera to the Raycaster constructor

**Comment by nzjony** [comment]
@heremaps-bot  retest this please
